src/server/ua_subscription_events.c

```diff
@@ -678,4 +678,216 @@ UA_Server_triggerEvent(UA_Server *server, const UA_NodeId eventNodeId,
+            if(!isNodeInTree_singleRef(
+                server, &eventFilter->selectClauses[i].typeDefinitionId,
+                &baseEventTypeId, UA_REFERENCETYPEINDEX_HASSUBTYPE)) {
+                result[i] = UA_STATUSCODE_BADTYPEDEFINITIONINVALID;
+                continue;
+            }
+            /* Check if attributeId is valid */
+            if(!((0 < eventFilter->selectClauses[i].attributeId) &&
+                 (eventFilter->selectClauses[i].attributeId < 28))) {
+                result[i] = UA_STATUSCODE_BADATTRIBUTEIDINVALID;
+                continue;
+            }
+            /* Check if browsePath contains null */
+            for(size_t j = 0; j < eventFilter->selectClauses[i].browsePathSize; ++j) {
+                if(UA_QualifiedName_isNull(
+                    &eventFilter->selectClauses[i].browsePath[j])) {
+                    result[i] = UA_STATUSCODE_BADBROWSENAMEINVALID;
+                    break;
+                }
+            }
+            if(result[i] != UA_STATUSCODE_GOOD)
+                continue;
+            /* Check if indexRange is defined */
+            if(!UA_String_equal(&eventFilter->selectClauses[i].indexRange,
+                                &UA_STRING_NULL)) {
+                /* Check if indexRange is parsable */
+                UA_NumericRange numericRange = UA_NUMERICRANGE("");
+                if(UA_NumericRange_parse(&numericRange,
+                                         eventFilter->selectClauses[i].indexRange) !=
+                   UA_STATUSCODE_GOOD) {
+                    result[i] = UA_STATUSCODE_BADINDEXRANGEINVALID;
+                    continue;
+                }
+                UA_free(numericRange.dimensions);
+                /* Check if attributeId is value */
+                if(eventFilter->selectClauses[i].attributeId != UA_ATTRIBUTEID_VALUE) {
+                    result[i] = UA_STATUSCODE_BADTYPEMISMATCH;
+                    continue;
+                }
+            }
+        }
+    }
+}
+
+UA_StatusCode
+UA_Event_staticWhereClauseValidation(UA_Server *server,
+                                     const UA_ContentFilter *contentFilter,
+                                     UA_ContentFilterResult * contentFilterResult) {
+    UA_ContentFilterResult_init(contentFilterResult);
+    contentFilterResult->elementResultsSize = contentFilter->elementsSize;
+    if(contentFilterResult->elementResultsSize == 0)
+        return UA_STATUSCODE_GOOD;
+    contentFilterResult->elementResults =
+        (UA_ContentFilterElementResult *)UA_Array_new(
+            contentFilterResult->elementResultsSize,
+            &UA_TYPES[UA_TYPES_CONTENTFILTERELEMENTRESULT]);
+    if(!contentFilterResult->elementResults)
+        return UA_STATUSCODE_BADOUTOFMEMORY;
+    for(size_t i = 0; i < contentFilterResult->elementResultsSize; ++i) {
+        UA_ContentFilterElementResult_init(&contentFilterResult->elementResults[i]);
```

Comment:
You have to write “ contentFilterResult->elementResults[i]” a million times.
Just create a pointer to it at the beginning of the loop and call it something short.
For example just “er” for element result.